### ISSUE TITLE
Buildbot.Orphis.net Automated Compiling

### ISSUE THREAD

**Issue by shinra358** [K4thos/Ikemen-GO-Plus#22]
Is it possible for you guys to use this site for automatic compiling? I would like to get back into mugen, yet I can't try this out because it is not compiled.

**Comment by shinlucho** [comment]
Well yea, it's annoying to download all the dependencies and then compile it only to see if it's what someone expected from that version or commit.

I'm not quite sure about that page, however we can use another one. 

That page only mentions SVN repositories but we use GIT, and it doesn't explain anything how it works or how to integrate it with an existing repository or anything.

I was thinking on [Travis CI](https://travis-ci.com/) to automatize testing each time we upload some new code, but it can also be used to build a executable if everything is ok.

Let me investigate this a bit and later I'll comment how to implement this.

**Comment by shinra358** [comment]
Until then, what dependencies do I need? I don't see that info anywhere.

**Comment by shinlucho** [comment]
I've added a readme with steps to compile Ikemen. Let me know if you have any problems compiling it.

Regarding autobuild feature, [AppVeyor](https://github.com/marketplace/appveyor) what we want, but isn't so user friendly. [RPCS3](https://rpcs3.net/) uses this service, but I think they parse AppVeyor on their website to [show it nicely](https://rpcs3.net/compatibility?b). I'll keep searching a bit.

**Comment by shinra358** [comment]
E:\Users\T Dean Jr\My Documents\Games\CREATION\Ikemen-GO-Plus-K4-experimental>se
t GOPATH=E:\Users\T Dean Jr\My Documents\Games\CREATION\Ikemen-GO-Plus-K4-experi
mental/go

```
E:\Users\*\My Documents\Games\CREATION\Ikemen-GO-Plus-K4-experimental>se
t CGO_ENABLED=1

E:\Users\*\My Documents\Games\CREATION\Ikemen-GO-Plus-K4-experimental>go
 build -ldflags -H=windowsgui -o Ikemen_GO.exe ./src
'go' is not recognized as an internal or external command,
operable program or batch file.

E:\Users\*\My Documents\Games\CREATION\Ikemen-GO-Plus-K4-experimental>pa
use
Press any key to continue . . .
```

**Comment by shinra358** [comment]
Had to restart. That wasn't in the tutorial. But afterward I get this:


```
E:\Users\*\My Documents\Games\CREATION\Ikemen-GO-Plus-K4-experimental>se
t GOPATH=E:\Users\*\My Documents\Games\CREATION\Ikemen-GO-Plus-K4-experi
mental/go

E:\Users\*\My Documents\Games\CREATION\Ikemen-GO-Plus-K4-experimental>se
t CGO_ENABLED=1

E:\Users\*\My Documents\Games\CREATION\Ikemen-GO-Plus-K4-experimental>go
 build -ldflags -H=windowsgui -o Ikemen_GO.exe ./src
src\font.go:5:2: cannot find package "github.com/go-gl/gl/v2.1/gl" in any of:
        C:\Go\src\github.com\go-gl\gl\v2.1\gl (from $GOROOT)
        E:\Users\*\My Documents\Games\CREATION\Ikemen-GO-Plus-K4-experim
ental\go\src\github.com\go-gl\gl\v2.1\gl (from $GOPATH)
src\input.go:5:2: cannot find package "github.com/go-gl/glfw/v3.2/glfw" in any o
f:
        C:\Go\src\github.com\go-gl\glfw\v3.2\glfw (from $GOROOT)
        E:\Users\*\My Documents\Games\CREATION\Ikemen-GO-Plus-K4-experim
ental\go\src\github.com\go-gl\glfw\v3.2\glfw (from $GOPATH)
src\sound.go:6:2: cannot find package "github.com/jfreymuth/go-vorbis/ogg/vorbis
" in any of:
        C:\Go\src\github.com\jfreymuth\go-vorbis\ogg\vorbis (from $GOROOT)
        E:\Users\*\My Documents\Games\CREATION\Ikemen-GO-Plus-K4-experim
ental\go\src\github.com\jfreymuth\go-vorbis\ogg\vorbis (from $GOPATH)
src\sound.go:7:2: cannot find package "github.com/timshannon/go-openal/openal" i
n any of:
        C:\Go\src\github.com\timshannon\go-openal\openal (from $GOROOT)
        E:\Users\*\My Documents\Games\CREATION\Ikemen-GO-Plus-K4-experim
ental\go\src\github.com\timshannon\go-openal\openal (from $GOPATH)
src\main.go:7:2: cannot find package "github.com/yuin/gopher-lua" in any of:
        C:\Go\src\github.com\yuin\gopher-lua (from $GOROOT)
        E:\Users\*\My Documents\Games\CREATION\Ikemen-GO-Plus-K4-experim
ental\go\src\github.com\yuin\gopher-lua (from $GOPATH)

E:\Users\*\My Documents\Games\CREATION\Ikemen-GO-Plus-K4-experimental>pa
use
Press any key to continue . . .
```

**Comment by shinra358** [comment]
Searched an error and a user had the same problem on mugenguild. So I saw directions that K4thos made and I noticed a couple things were missing from the readme:

```
4. Download this unofficial soft OpenAL dll file: https://www.dropbox.com/s/0obp5tmiy64wtl8/OpenAL32.lib?dl=1
and place it within C:\TDM-GCC-64\lib directory (for some reason engine can't be compiled with official OpenAL dev package)
```
This was stated but it specifically said the 32 bit file not the 64 bit file.

...
5. Install git for windows (required by one of the libraries): https://npm.taobao.org/mirrors/git-for-windows/v2.15.1.windows.2/
...

This says required but the readme states optional.

...
6. Use get.sh shell file or get.bat batch file to download libraries used by Ikemen GO
...
There isn't a get.bat file provided.
